Disconnect when deleting Item (Fabric)

When i try deleting an Item on the Right Panel Minecraft Disconnects me because of this exception:

[Video](https://www.youtube.com/watch?v=m8sKSaYGYWg)

Fabric API Version: 0.97.8+1.20.6
Fabric Loader: 0.15.10

No other Mods where installed

```console
[12:41:00] [Netty Local Client IO #0/ERROR]: Error sending packet serverbound/minecraft:custom_payload
io.netty.handler.codec.EncoderException: Failed to encode packet 'serverbound/minecraft:custom_payload' (emi:create_item)
	at net.minecraft.class_9136.handler$zha000$fabric-networking-api-v1$encode(class_9136.java:547) ~[client-intermediary.jar:?]
	at net.minecraft.class_9136.method_56426(class_9136.java:55) ~[client-intermediary.jar:?]
	at net.minecraft.class_9136.encode(class_9136.java:14) ~[client-intermediary.jar:?]
	at net.minecraft.class_2545.method_10838(class_2545.java:26) ~[client-intermediary.jar:?]
	at net.minecraft.class_2545.encode(class_2545.java:12) ~[client-intermediary.jar:?]
	at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[netty-codec-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113) ~[netty-transport-4.1.97.Final.jar:?]
	at net.minecraft.class_2535$2.write(class_2535.java:518) ~[client-intermediary.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:940) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:966) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:934) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:984) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1025) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:306) ~[netty-transport-4.1.97.Final.jar:?]
	at net.minecraft.class_2535.method_36942(class_2535.java:342) ~[client-intermediary.jar:?]
	at net.minecraft.class_2535.method_52917(class_2535.java:337) ~[client-intermediary.jar:?]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.97.Final.jar:?]
	at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54) ~[netty-transport-4.1.97.Final.jar:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.97.Final.jar:?]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.97.Final.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: io.netty.handler.codec.EncoderException: Empty ItemStack not allowed
	at net.minecraft.class_1799$2.method_57387(class_1799.java:173) ~[client-intermediary.jar:?]
	at net.minecraft.class_1799$2.encode(class_1799.java:160) ~[client-intermediary.jar:?]
	at dev.emi.emi.network.CreateItemC2SPacket.write(CreateItemC2SPacket.java:26) ~[emi-1.1.5+1.20.6+fabric.jar:?]
	at dev.emi.emi.platform.fabric.EmiMainFabric.lambda$registerPacketReader$2(EmiMainFabric.java:47) ~[emi-1.1.5+1.20.6+fabric.jar:?]
	at net.minecraft.class_9139$1.encode(class_9139.java:25) ~[client-intermediary.jar:?]
	at net.minecraft.class_8710$1.method_56489(class_8710.java:52) ~[client-intermediary.jar:?]
	at net.minecraft.class_8710$1.method_56490(class_8710.java:57) ~[client-intermediary.jar:?]
	at net.minecraft.class_8710$1.encode(class_8710.java:39) ~[client-intermediary.jar:?]
	at net.minecraft.class_9139$8.encode(class_9139.java:81) ~[client-intermediary.jar:?]
	at net.minecraft.class_9139$9.method_56442(class_9139.java:97) ~[client-intermediary.jar:?]
	at net.minecraft.class_9139$9.encode(class_9139.java:87) ~[client-intermediary.jar:?]
	at net.minecraft.class_9136.method_56426(class_9136.java:53) ~[client-intermediary.jar:?]
	... 24 more
[12:41:00] [Server thread/INFO]: xStopho lost connection: Disconnected
[12:41:00] [Server thread/INFO]: xStopho left the game
[12:41:00] [Server thread/INFO]: Stopping singleplayer server as player logged out
[12:41:00] [Server thread/INFO]: Stopping server
[12:41:00] [Server thread/INFO]: Saving players
[12:41:00] [Server thread/INFO]: Saving worlds
[12:41:00] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[12:41:00] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[12:41:00] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[12:41:00] [Server thread/INFO]: ThreadedAnvilChunkStorage (New World): All chunks are saved
[12:41:00] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[12:41:00] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[12:41:00] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[12:41:01] [Render thread/WARN]: Client disconnected with reason: Internal Exception: io.netty.handler.codec.EncoderException: Failed to encode packet 'serverbound/minecraft:custom_payload' (emi:create_item)
```

--- a/xplat/src/main/java/dev/emi/emi/network/CreateItemC2SPacket.java
+++ b/xplat/src/main/java/dev/emi/emi/network/CreateItemC2SPacket.java
@@ -17,13 +17,13 @@ public class CreateItemC2SPacket implements EmiPacket {
 	}
 
 	public CreateItemC2SPacket(RegistryByteBuf buf) {
-		this(buf.readByte(), ItemStack.PACKET_CODEC.decode(buf));
+		this(buf.readByte(), ItemStack.OPTIONAL_PACKET_CODEC.decode(buf));
 	}
 
 	@Override
 	public void write(RegistryByteBuf buf) {
 		buf.writeByte(mode);
-		ItemStack.PACKET_CODEC.encode(buf, stack);
+		ItemStack.OPTIONAL_PACKET_CODEC.encode(buf, stack);
 	}
 
 	@Override

--- a/xplat/src/main/java/dev/emi/emi/network/FillRecipeC2SPacket.java
+++ b/xplat/src/main/java/dev/emi/emi/network/FillRecipeC2SPacket.java
@@ -52,7 +52,7 @@ public class FillRecipeC2SPacket implements EmiPacket {
 		int size = buf.readVarInt();
 		stacks = Lists.newArrayList();
 		for (int i = 0; i < size; i++) {
-			stacks.add(ItemStack.PACKET_CODEC.decode(buf));
+			stacks.add(ItemStack.OPTIONAL_PACKET_CODEC.decode(buf));
 		}
 	}
 
@@ -73,7 +73,7 @@ public class FillRecipeC2SPacket implements EmiPacket {
 		}
 		buf.writeVarInt(stacks.size());
 		for (ItemStack stack : stacks) {
-			ItemStack.PACKET_CODEC.encode(buf, stack);
+			ItemStack.OPTIONAL_PACKET_CODEC.encode(buf, stack);
 		}
 	}
 

